solver/src/liquidity.rs

```diff
@@ -104,6 +107,12 @@ impl Default for LimitOrder {
     }
 }
 
+// pub struct AmmOrder<T> {
+//     pub reserves: HashMap<H160, T>,
+//     pub fee: Ratio<u32>,
+//     pub settlement_handling: Arc<dyn SettlementHandling<Self>>,
+// }
+
```

Comment:
Is this a template for amm orders? If so, an explanation comment would be nice, on the other hand I don't see the need.

Reply:
This is an idea I was tinkering with that we could go with instead of having both weighted and constant product orders. I would actually like to go with this at some point but haven't had the time to put it together. The idea is that eventually we will use other forms of liquidity like 0x orders which aren't pools but rather just considered another form of Amm. This is closely related to what is happening in #769 and the discussion I've been having with @twalth3r there. From the perspective of the solver team only having to become compatible with one breaking change this would accommodate both. Unfortunately it relies heavily on whether we go with the term "amms" or "pools" in the "PoolModel" or possibly "AmmModel" and abandon the term pools, in favour of amms, in a few places. 

Same goes for uniswap v3 - if we are going to call them pools. 

If others have any further insight I would be happy to continue this discussion. Overall it boils down to whether or not the solver team will have to adapt to the keys of the dictionary they are receiving although I suppose they would have to adapt to these other forms of liquidity regardless.

Reply:
I will cut it out of here for now and put it back when I have a chance.